tests/test_dicom/test_dicom_anonymise.py

```diff
@@ -30,13 +28,16 @@ def test_is_anonymised_dataset():
     ds_anon_manual = ds
     ds_anon_manual.PatientName = "Anonymous"
 
+
     ds_anon_func = anonymise_dataset(ds, delete_private_tags=False,
                                      copy_dataset=True)
 
     # Patient Name should've been the only non-anonymous value. If not,
     # test data has been changed
     for elem_manual, elem_func in zip(ds_anon_manual, ds_anon_func):
-        assert elem_manual.value == elem_func.value
+        # Assumes all non-private, identifying tags are top-level
+        if elem_manual.VR != 'SQ':
+            assert elem_manual == elem_func
```

Comment:
Ignores sequences for now. Hunch for the moment is that floating point numbers are present in the sequences, so testing for equality needs to be a touch looser (allclose instead of equal).

Thinking of writing a function that asserts two DICOM datasets are equal by looping through every element, elements in sequence, element in sequence in sequence, etc., checking for type and running an `allclose()` for any floating points...

Reply:
This might differ between 1.2.2 and 1.2.0 because `__eq__()` might not check as deeply as it did before...

Yet to verify though